src/parse/decl.rs

```diff
@@ -1628,6 +1672,16 @@ impl Declarator {
 }
 
 impl Type {
+    fn type_len(&self) -> usize {
+        match self {
+            ty if ty.is_scalar() => 1,
+            Type::Array(_, ArrayType::Fixed(size)) => *size as usize,
+            Type::Array(_, ArrayType::Unbounded) => 0, // TODO: is this reasonable?
```

Comment:
I would have used `.unwrap_or(0)` if it returned `Option`. Since this is an internal API, I'm ok with throwing away a little clarity for convenience, especially since it's only used in one place.